rero_ils/modules/items/api/record.py

```diff
@@ -24,25 +24,48 @@
 from ...libraries.api import Library
 from ...locations.api import Location
 from ...organisations.api import Organisation
-from ...utils import generate_item_barcode, trim_barcode_for_record
+from ...utils import extracted_data_from_ref, generate_item_barcode, \
+    get_ref_for_pid, trim_barcode_for_record
 
 
 class ItemRecord(IlsRecord):
     """Item record class."""
 
+    def extended_validation(self, **kwargs):
+        """Add additional record validation.
+
+        Ensures that item of type issue is created only on
+        holdings of type serials.
+
+        Ensures that item of type issue has the issue field.
+
+        Ensures that standard item has no issue field.
+
+        :returns: False if
+            - holdings type is not journal and item type is issue.
+            - item type is journal and field issue exists.
+            - item type is standard and field issue does not exists.
+        """
+        from ...holdings.api import Holding
+        holding = Holding.get_record_by_pid(self.holding_pid)
+        is_serial = holding.holdings_type == 'serial'
+        if is_serial and self.get('type') == 'standard':
+            return False
+        issue = self.get('issue', {})
```

Comment:
Could we write this logic into the json schema?
If the `type` is `issue` we need a key `issue` as well.

Reply:
will be added in next PR